src/credentials/activation.ts

```diff
@@ -44,7 +45,9 @@ export async function loginWithMostRecentCredentials(
             credentialType: SharedCredentialsProvider.getCredentialsType(),
             credentialTypeId: previousCredentialsId,
         }
-        if ((await manager.getCredentialsProvider(loginCredentialsId))!.canAutoConnect()) {
+        const provider: CredentialsProvider | undefined = await manager.getCredentialsProvider(loginCredentialsId)
```

Comment:
type inference should make the explicit type unnecessary (without any loss of type-checking):
```suggestion
        const provider = await manager.getCredentialsProvider(loginCredentialsId)
```